lib/src/dart2_suggestors/pubspec_over_react_upgrader.dart

```diff
@@ -71,25 +71,32 @@ class PubspecOverReactUpgrader implements Suggestor {
     if (overReactMatch != null) {
       // over_react is already in pubspec.yaml
       final constraintValue = overReactMatch.group(2);
-      final constraint = VersionConstraint.parse(constraintValue);
-
-      if (shouldUpdateVersionRange(
-          targetConstraint: targetConstraint,
-          constraint: constraint,
-          shouldIgnoreMin: shouldIgnoreMin)) {
-        // Wrap the new constraint in quotes if required.
-        var newValue =
-            generateNewVersionRange(constraint, targetConstraint).toString();
-
-        if (mightNeedYamlEscaping(newValue)) {
-          newValue = '"$newValue"';
+      try {
+        final constraint = VersionConstraint.parse(constraintValue);
+
+        if (shouldUpdateVersionRange(
+            targetConstraint: targetConstraint,
+            constraint: constraint,
+            shouldIgnoreMin: shouldIgnoreMin)) {
+          // Wrap the new constraint in quotes if required.
+          var newValue =
+              generateNewVersionRange(constraint, targetConstraint).toString();
+
+          if (mightNeedYamlEscaping(newValue)) {
+            newValue = '"$newValue"';
+          }
+
+          // Update the version constraint to ensure a safe minimum bound.
+          yield Patch(
+              sourceFile,
+              sourceFile.span(overReactMatch.start, overReactMatch.end),
+              '  over_react: $newValue');
         }
-
-        // Update the version constraint to ensure a safe minimum bound.
-        yield Patch(
-            sourceFile,
-            sourceFile.span(overReactMatch.start, overReactMatch.end),
-            '  over_react: $newValue');
+      } catch (e) {
+        // We can skip these they are versions we dont want to mess with in this codemod
+        if (e.toString().contains('git:') || e.toString().contains('path:'))
+          return;
+        rethrow;
```

Comment:
You sure you want to rethrow? Won't that result in the codemod choking out when it finds one of these?

Reply:
So if `git:` or `path:` show up in the error string, we expect those errors and don't want to do anything so we return on them; Otherwise, its a non expected error and so i rethrow, since we don't know if its expected or not.

Reply:
That does bring up the possibility of adding the return statement to line 97. Effective Dart [points out](https://dart.dev/guides/language/effective-dart/style#do-use-curly-braces-for-all-flow-control-structures) that if you don't use curly braces that the `if` should be on a single line, which I do think would make it more readable to the effect that you're going for! (Sorry I didn't catch that on my pass!)

Reply:
blame ddev format